### PR TITLE
update url hash to match selected leaf, select leaf on load based on hash

### PR DESCRIPTION
**Pull request by rlskoeser**
## testing notes
test the following interactions with the timetree on the render PR site: https://lenape-timetree-dev-pr-96.onrender.com/
- [x] selecting a leaf changes the url in the browser location (should add a hash `#xxx` based on the slugified version of the leaf title)
- [x] closing the leaf details with the "X" removes the leaf id from the browser location (should be just `#` or `#`)
- [x] selecting and deselecting leaves should not trigger entries in browser history (i.e., if you select and close multiple leaves, when you use the browser back button the url should not page back through the leaves you've selected)
- [x] loading the page with a leaf hash id in the url should show the leaf details; can test by selecting a leaf, copying the url, and then opening in a new browser tab or window

### CODE REVIEW

**Comment by jhimpele** [comment]
selecting and deselecting leaves should not trigger entries in browser history - back button is not available, but the browser history does record the series of selected leaves.

File: assets/js/leaves.js [hunk 1]
```diff
@@ -50,6 +50,11 @@ class Leaf {
     Array.from(selected).forEach((item) => {
       item.classList.remove(Leaf.selectedClass);
     });
+
+    // change url location hash to indicate no leaf is selected
+    // needs to be non-empty to avoid page reload
+    window.location.replace("#");
+    history.replaceState(null, "", window.hash);
```
Comment (quadrismegistus) [inline assets/js/leaves.js:57]: I really like when websites do this. Every object/view ideally should have a stable URL. Makes citation a lot more powerful. So: bravo!

File: assets/js/leaves.js [hunk 2]
```diff
@@ -79,6 +84,11 @@ class Leaf {
       item.classList.add(Leaf.selectedClass);
     }
 
+    // update URL to reflect the currently selected leaf;
+    // replace the location & state to avoid polluting browser history
+    window.location.replace(`#${target.getAttribute("data-id")}`);
+    history.replaceState(null, "", window.hash);
+
```
Comment (quadrismegistus) [inline assets/js/leaves.js:91]: see above

File: assets/js/timetree.js [hunk 3]
```diff
@@ -81,7 +81,6 @@ let sortedLeaves = data.leaves
   .sort((a, b) => a.sort_date > b.sort_date);
 // use url as id for node in graph; set type to leaf; set century
 sortedLeaves.forEach((leaf) => {
-  leaf.id = leaf.url;
   leaf.type = "leaf";
```
Comment (quadrismegistus) [inline assets/js/timetree.js:84]: what is `leaf.id` now then?

**Review by quadrismegistus — APPROVED** [review]
Looks great! Love the feature

File: assets/js/timetree.js (hunk 3 above)
Comment (rlskoeser) [reply assets/js/timetree.js:84]: short id / slug; worth commenting?

**Pull request merged**